.changelog/2419.txt

```diff
@@ -0,0 +1,4 @@
+```release-note:bug
+`resource/kubernetes_daemon_set_v1`: fix an issue with the provider not waiting for rollout with `wait_for_rollout = true`.
```

Comment:
I would suggest a mention of the change in behaviour here, so users are well advised of what to expect when upgrading.

Reply:
@alexsomesan That's a great point. I'm not sure about the wording of the note. I'm thinking about adding the following to the existing entry: "Note: As the `wait_for_rollout` is `true` by default, users might experience the `apply` operation of the existing code taking longer." 

Or should the `UPGRADE NOTES` section be used instead as I see it was used in past e.g. https://github.com/hashicorp/terraform-provider-kubernetes/blob/main/CHANGELOG.md#161-april-18-2019 . In this case, I'm not sure how to add this extra section.

Reply:
If it is undesirable to change the semantics of the existing `wait_for_rollout` attribute (due to potential breakage of existing code), perhaps a new `really_wait_for_rollout` attribute can be introduced instead, and the old one [deprecated](https://developer.hashicorp.com/terraform/plugin/framework/deprecations).